lib/perl/Genome/Model/Tools/Tcga/Sdrf.pm

```diff
@@ -197,21 +198,30 @@ sub fill_in_common_fields {
 }
 
 sub resolve_cghub_id {
-    my $self = shift;
-    my $build = shift;
+    my ($self, $build) = @_;
 
     unless (defined $CGHUB_INFO) {
         $CGHUB_INFO = $self->load_cghub_info("BAM_path");
     }
 
+    my $id_by_bam_base;
+    while (my($bam_path, $cghub_id) = each %$CGHUB_INFO) {
+        $id_by_bam_base->{basename $bam_path} = $cghub_id;
+    }
+
     unless (defined $CGHUB_INFO_BY_TCGA_NAME) {
         $CGHUB_INFO_BY_TCGA_NAME = $self->load_cghub_info("TCGA_Name");
     }
-    my $id = $CGHUB_INFO->{$build->whole_rmdup_bam_file};
+
+    my $merged_bam = $build->whole_rmdup_bam_file;
+    my $id = $CGHUB_INFO->{$merged_bam};
     unless (defined $id) {
-        $id = $CGHUB_INFO_BY_TCGA_NAME->{$build->subject->extraction_label};
+        $id = $CGHUB_INFO_BY_TCGA_NAME->{$build->subject->name_in_vcf};
         unless (defined $id) {
-            die("CGHub id could not be resolved for build ".$build->id." with bam file ".$build->whole_rmdup_bam_file);
+            $id = $id_by_bam_base->{basename $merged_bam};
```

Comment:
Should this be the first thing tried?  Then we wouldn't have to try $id=$CGHUB_INFO->{$merged_bam} and would only have two 'unlesses' instead of three